plugin/storage/es/spanstore/reader.go

```diff
@@ -159,6 +161,20 @@ func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
 	return timeRangeIndices
 }
 
+func getSourceFn(archive bool, maxNumSpans int) sourceFn {
+	return func(query elastic.Query, nextTime uint64) *elastic.SearchSource {
+		s := elastic.NewSearchSource().
+			Query(query).
+			Size(defaultDocCount).
+			TerminateAfter(maxNumSpans).
+			Sort("startTime", true)
+		if !archive {
+			s.SearchAfter(nextTime)
```

Comment:
This query uses scroll API https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-search-after.html.

`SearchAfter` says to include results with `startTime` parameter after `nextTime`. In archive storage we want to include all results.